### PR TITLE
added new test cases for remotesaveprovider

### PR DESCRIPTION
**Pull request by v-vbhanndari**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -731,11 +731,13 @@
     ]
   },
   "devDependencies": {
+    "@istanbuljs/nyc-config-typescript": "1.0.2",
```
Comment (BidishaMS) [inline package.json:734]: undo this change. Keep it where it was early. Unnecessary diff changes it shows

File: package.json [hunk 2]
```diff
@@ -731,11 +731,13 @@
...
     "@types/chai": "^4.2.14",
     "@types/fs-extra": "^9.0.11",
     "@types/glob": "^7.1.3",
     "@types/mocha": "^8.0.3",
     "@types/node": "^14.14.6",
+    "@types/node-fetch": "^2.6.2",
```
Comment (BidishaMS) [inline package.json:740]: Should'nt this be added in dependencies too?

File: src/web/client/dal/remoteSaveProvider.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@
  * Licensed under the MIT License. See License.txt in the project root for license information.
  */
 
+import fetch, { RequestInit } from "node-fetch";
```
Comment (BidishaMS) [inline src/web/client/dal/remoteSaveProvider.ts:6]: @tyaginidhi , Vipin has used node-fetch instead of vanilla JS fetch method. Few reasons:
1. It is a light weight NPM package which is similar to JS fetch.
2. JS fetch mocking was not achievable so need to switch to node-fetch.

File: src/web/client/test/integration/remoteSaveProvider.test.ts [hunk 4]
```diff
@@ -0,0 +1,634 @@
...
+    entityAttributesWithBase64Encoding,
+} from "../../schema/constants";
+import * as schemaHelperUtil from "../../utilities/schemaHelperUtil";
+import WebExtensionContext from "../../../client/WebExtensionContext";
+import { getHeader } from "../../common/authenticationProvider";
+import { httpMethod, BAD_REQUEST } from "../../common/constants";
+import * as errorHandler from "../../common/errorHandler";
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+import { expect } from "chai";
+describe("remoteSaveProvider", () => {
```
Comment (BidishaMS) [inline src/web/client/test/integration/remoteSaveProvider.test.ts:23]: newline

File: src/web/client/test/integration/remoteSaveProvider.test.ts [hunk 5]
```diff
@@ -0,0 +1,634 @@
+/*
```
Comment (BidishaMS) [inline src/web/client/test/integration/remoteSaveProvider.test.ts:1]: Move this to Unit folder.

**Comment by BidishaMS** [comment]
Please add a coverage report

File: src/web/client/test/integration/remoteSaveProvider.test.ts (hunk 4 above)
Comment (v-vbhanndari) [reply src/web/client/test/integration/remoteSaveProvider.test.ts:23]: Done

File: src/web/client/dal/remoteSaveProvider.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (v-vbhanndari) [reply src/web/client/dal/remoteSaveProvider.ts:6]: Ok. Please let me know how to do so.

File: src/web/client/test/integration/remoteSaveProvider.test.ts (hunk 5 above)
Comment (v-vbhanndari) [reply src/web/client/test/integration/remoteSaveProvider.test.ts:1]: Not working

**Comment by v-vbhanndari** [comment]
As i raised the new PR for the same because there are lots of changes happen before merge this branch
